cli/webhook.py

```diff
@@ -0,0 +1,125 @@
+                        self.client.notify(channel=channel, username=self.username,
+                                           icon_emoji=self.emoji, text=text)
+
+    def invoke_webhook(self, appdata, hook_input_metric):
+        self.webhookSetting()
+        try:
+            action = app_name = envr = user = ''
+            temp = hook_input_metric.split(',')
+            for var in temp:
+                varAnother = var.split('=')[0]
+                if varAnother == 'event':
+                    action = var.split('=')[1]
+                    continue
+
+                if varAnother == 'app_name':
+                    app_name = var.split('=')[1]
+                    continue
+
+                if varAnother == 'env':
+                    envr = var.split('=')[1]
+                    continue
+
+                if varAnother == 'user':
+                    user = var.split('=')[1]
+                    continue
+            if len(action) == 0 or len(app_name) == 0 or len(envr) == 0 or len(user) == 0:
+                raise ValueError
+
+        except (Exception, KeyError, ValueError, IndexError) as e:
+            print("The following error occurred: %s" %
+                  e)
+            raise
+        try:
+            if 'notifications' in appdata:
+                channelsSet = Set(appdata['notifications']['channels'])
+                envSet = Set(appdata['notifications']['envs'])
+                commandsSet = Set(appdata['notifications']['commands'])
+                # to handle no tag at all
+                if (len(channelsSet) == 0 or len(envSet) == 0 or len(commandsSet) == 0):
+                    print("Notificaton tag missing. Not posting message to slack!")
+                    return
+            else:
+                print("Notificaton tag missing. Not posting message to slack!")
+                return
+            # to handle all tag for env and commands
+            if list(envSet)[0] == 'all':
+                envSet = ['dev', 'production', 'staging', 'local']
+
+            if list(commandsSet)[0] == 'all':
+                commandsSet = ['pull', 'build', 'push']
+        except (Exception, KeyError, ValueError) as e:
+            # notify to channel and log it as well
+            print("The following error occurred: %s" %
+                  e)
+            raise
+        try:
+            function_execution_start_time = datetime.now()
+            if ('post' in action and envr in envSet and action.split('_')[1] in commandsSet):
+                for channel in channelsSet:
+                    slackMessage = ("Completed *" + action.split('_')[1] + "* of *" + app_name +
+                                    "* on *" + envr + "* in *" +
+                                    str((datetime.now() - function_execution_start_time)
+                                        .total_seconds() * 1000) +
+                                    "* miliseconds (triggered by *" + user + "*)")
```

Comment:
Can you make the difference human readable? Add milliseconds to the lambda function [here](http://stackoverflow.com/questions/6574329/how-can-i-produce-a-human-readable-difference-when-subtracting-two-unix-timestam).